rules/integrations/aws/discovery_ec2_multiple_discovery_api_calls_via_cli.toml

```diff
@@ -0,0 +1,147 @@
+
+### Response and Remediation
+
+- **Confirm Authorized Access**: If the discovery activity appears unauthorized, consider immediate steps to restrict the user or resource’s permissions.
+- **Review and Remove Unauthorized API Calls**: If the actor is not authorized to perform discovery actions, investigate and potentially disable their permissions or access keys to prevent further misuse.
+- **Enhance Monitoring for Discovery Patterns**: Consider additional logging or alerting for high-frequency discovery API calls, especially if triggered from new or unrecognized resources.
+- **Policy Review and Updates**: Review IAM policies associated with the actor, ensuring restrictive permissions and MFA enforcement where possible to prevent unauthorized discovery.
+
+### Additional Information
+
+For further guidance on AWS infrastructure discovery and best practices, refer to [AWS CloudTrail documentation](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-event-reference.html) and MITRE ATT&CK’s [Cloud Infrastructure Discovery](https://attack.mitre.org/techniques/T1580/).
+"""
+references = ["https://stratus-red-team.cloud/attack-techniques/AWS/aws.discovery.ec2-enumerate-from-instance/"]
+risk_score = 21
+rule_id = "74f45152-9aee-11ef-b0a5-f661ea17fbcd"
+severity = "low"
+tags = [
+    "Domain: Cloud",
+    "Data Source: AWS",
+    "Data Source: AWS EC2",
+    "Data Source: AWS IAM",
+    "Data Source: AWS S3",
+    "Use Case: Threat Detection",
+    "Tactic: Discovery",
+]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
+from logs-aws.cloudtrail*
+
+// create time window buckets of 10 seconds
+| eval time_window = date_trunc(10 seconds, @timestamp)
+| where
+    event.dataset == "aws.cloudtrail"
+
+    // filter on CloudTrail audit logs for IAM, EC2, and S3 events only
+    and event.provider in ("iam.amazonaws.com", "ec2.amazonaws.com", "s3.amazonaws.com")
+
+    // ignore AWS service actions
+    and aws.cloudtrail.user_identity.type != "AWSService"
+
+    // filter for aws-cli specifically
+    and user_agent.name == "aws-cli"
+
+    // exclude DescribeCapacityReservations events related to AWS Config
+    and not event.action in ("DescribeCapacityReservations")
+
+    // filter for Describe and List API calls
+    and (
+            event.action like "Describe*" or
+            event.action like "List*" or
+            event.action like "Get*"
+    )
+
+// extract owner, identity type, and actor from the ARN
+| dissect aws.cloudtrail.user_identity.arn "%{}::%{owner}:%{identity_type}/%{actor}"
+| eval
+
+    // filter out AWS, Elastic, Datadog, and Wiz service roles by ARN
+    aws_vendor = starts_with(actor, "AWSServiceRoleForConfig"),
+    elastic_vendor = starts_with(actor, "Elastic"),
+    datadog_vendor = starts_with(actor, "DataDog"),
+    wiz_vendor = starts_with(actor, "Wiz")
+| where
+    aws_vendor != true and
+    elastic_vendor != true and
+    datadog_vendor != true and
+    wiz_vendor != true
```

Comment:
Should we leave these type of exclusions up to users?

Reply:
Agreed, as discussed I will remove them, however specifically ignore the AWSServiceRoleForConfig as we can be specific about the ARN path which cannot be duplicated but is known to generate FPs.